src/hotspot/share/code/nmethod.hpp

```diff
@@ -491,6 +493,8 @@ class nmethod : public CodeBlob {
 #endif
   );
 
+  static nmethod* replace_nmethod(nmethod* nm, int comp_level_override=-1);
```

Comment:
I think we need a function with the name reflecting our purpose:
```
 // Relocate the nmethod to the code heap identified by code_blob_type.
 // Returns nullptr if the code heap does not have enough space, otherwise
 // the relocated nmethod. The original nmethod will be invalidated.
 // If nm is already in the needed code heap, it is not relocated and the function returns it.
 static nmethod* relocate_to(nmethod* nm, CodeBlobType code_blob_type);
```